### PR TITLE
Port CBT

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Port of https://github.com/tgstation/tgstation/pull/55373 

This is monkey code, i just copied out the PR and it's working. Can probably cause some issues, though i had no problem on my part

Also change our linters to the CI suit, it's a bit more intransigeant but now we are in par with tgstation

need https://github.com/tgstation/TerraGov-Marine-Corps/pull/6680 to compile

### CODE REVIEW

**Comment by BraveMole** [comment]
@spookydonut Probably will need your help to make this pass tests

File: code/controllers/subsystem/vote.dm [hunk 1]
```diff
@@ -205,6 +205,8 @@ SUBSYSTEM_DEF(vote)
 					to_chat(usr, "<span class='warning'>The next ground map has already been selected.</span>")
 					return FALSE
 				var/list/maps = list()
+				if(!config.maplist)
+					return
```
Comment (BraveMole) [inline code/controllers/subsystem/vote.dm:209]: All of this is not needed for CBT, but why not

File: code/datums/map_config.dm [hunk 2]
```diff
@@ -61,11 +62,18 @@
...
 		if(error_if_missing)
 			log_world("map_config not found: [filename]")
-		return
+		if(!load_default)
+			return
+		switch(maptype)
+			if(GROUND_MAP)
+				return LoadConfig("_maps/lv624.json", error_if_missing, maptype)
+			if(SHIP_MAP)
+				return LoadConfig("_maps/pillar_of_spring.json", error_if_missing, maptype)
```
Comment (BraveMole) [inline code/datums/map_config.dm:75]: next_map.json is not on the git repo, so integration test couldn't load it. Added that as a backup

File: code/datums/map_config.dm [hunk 3]
```diff
@@ -175,7 +184,8 @@
 	else
 		for(var/a in subtypesof(/datum/game_mode))
 			var/datum/game_mode/G = a
-			gamemodes += initial(G.config_tag)
+			if(initial(G.config_tag))
+				gamemodes += initial(G.config_tag)
```
Comment (BraveMole) [inline code/datums/map_config.dm:188]: Not usefull for CBT, remove runtime when config has failed

File: tgui/.gitignore [hunk 4]
```diff
@@ -13,8 +13,8 @@ package-lock.json
 
 ## Build artifacts
 /public/.tmp/**/*
-##/public/**/*
-##!/public/*.html
+/public/**/*
+!/public/*.html
```
Comment (BraveMole) [inline tgui/.gitignore:17]: Not persuaded this is usefull, but it's like that in TG

Comment (SplinterGP) [reply tgui/.gitignore:17]: thats gitignore stuff

File: code/controllers/subsystem/vote.dm (hunk 1 above)
Comment (TiviPlus) [reply code/controllers/subsystem/vote.dm:209]: Then why did you not atomize it

File: code/datums/map_config.dm (hunk 2 above)
Comment (TiviPlus) [inline code/datums/map_config.dm:75]: this should be atomized

File: tools/ci/check_grep.sh [hunk 5]
```diff
@@ -6,119 +6,80 @@ shopt -s globstar
 
 st=0
 
-echo "Checking for TGM formatting"
 if grep -El '^\".+\" = \(.+\)' _maps/**/*.dmm;	then
-    echo "Non-TGM formatted map detected. Please convert it using Map Merger!"
+    echo "ERROR: Non-TGM formatted map detected. Please convert it using Map Merger!"
     st=1
 fi;
```
Comment (TiviPlus) [inline tools/ci/check_grep.sh:12]: wtf is all this

File: tools/ci/check_grep.sh [hunk 6]
```diff
@@ -6,119 +6,80 @@ shopt -s globstar
...
-echo "Checking long list formatting"
-if pcregrep -nM '^(\t)[\w_]+ = list\(\n\1\t{2,}' code/**/*.dm; then
-    echo "long list overidented, should be two tabs"
-    st=1
-fi;
-if pcregrep -nM '^(\t)[\w_]+ = list\(\n\1\S' code/**/*.dm; then
-    echo "long list underindented, should be two tabs"
-    st=1
-fi;
```
Comment (TiviPlus) [inline tools/ci/check_grep.sh:91]: ?????????

File: tools/ci/ci_config.txt [hunk 7]
```diff
@@ -1,7 +1,7 @@
 SQL_ENABLED
 ADDRESS 127.0.0.1
 PORT 3306
-FEEDBACK_DATABASE tgmc_ci
+FEEDBACK_DATABASE tg_ci
```
Comment (TiviPlus) [inline tools/ci/ci_config.txt:4]: ?????

**Review by TiviPlus — CHANGES_REQUESTED** [review]
DNM due to build changes

Why do you not delete the bundles

File: code/modules/clothing/head/misc.dm [hunk 8]
```diff
@@ -1,7 +1,7 @@
 
 
 /obj/item/clothing/head/centhat
-	name = "\improper CentComm. hat"
+	name = "\improper centcomm. hat"
```
Comment (BraveMole) [inline code/modules/clothing/head/misc.dm:4]:
```suggestion
	name = "\improper Centcomm. hat"
```

File: html/changelogs/changelog2016.html [hunk 9]
```diff
@@ -1871,7 +1871,7 @@ <h3 class='author'>Infernus44 updated:</h3>
 		<li class='bugfix'>Fixed wrong song being played for inflatable wall when hit</li>
 		<li class='bugfix'>Fixed huggers magically activating door controls</li>
 		<li class='bugfix'>Fixed lockers allowing aliens to hide inside</li>
-		<li class='bugfix'>Fixed larvas bursting on Centcomm level</li>
+		<li class='bugfix'>Fixed larvas bursting on centcomm level</li>
```
Comment (BraveMole) [inline html/changelogs/changelog2016.html:1874]:
```suggestion
		<li class='bugfix'>Fixed larvas bursting on Centcomm level</li>
```

File: html/changelogs/changelog2016.html [hunk 10]
```diff
@@ -2060,7 +2060,7 @@ <h3 class='author'>Rahlzel updated:</h3>
 	<h2 class='date'>15 February 2016</h2>
 	<h3 class='author'>Rahlzel updated:</h3>
 	<ul class='changes bgimages16'>
-		<li class='bugfix'>Logged-in humans at Centcomm and Thunderdome are now ignored when checking for alien victory.</li>
+		<li class='bugfix'>Logged-in humans at centcomm and Thunderdome are now ignored when checking for alien victory.</li>
```
Comment (BraveMole) [inline html/changelogs/changelog2016.html:2063]:
```suggestion
		<li class='bugfix'>Logged-in humans at Centcomm and Thunderdome are now ignored when checking for alien victory.</li>
```

File: html/changelogs/changelog20182.html [hunk 11]
```diff
@@ -563,7 +563,7 @@ <h3 class='author'>Surealistik updated:</h3>
 		</ul>
 		<h3 class='author'>Francinum updated:</h3>
 		<ul class='changes bgimages16'>
-			<li class='wip'>Brings CentComm slightly up to date, with a functional shuttle and a place for ERTs to return to.</li>
+			<li class='wip'>Brings centcomm slightly up to date, with a functional shuttle and a place for ERTs to return to.</li>
```
Comment (BraveMole) [inline html/changelogs/changelog20182.html:566]:
```suggestion
			<li class='wip'>Brings Centcomm slightly up to date, with a functional shuttle and a place for ERTs to return to.</li>
```

File: tgui/.gitignore (hunk 4 above)
Comment (BraveMole) [reply tgui/.gitignore:17]: Definitly 100% needed

File: code/modules/clothing/head/misc.dm [hunk 12]
```diff
@@ -1,7 +1,7 @@
 
 
 /obj/item/clothing/head/centhat
-	name = "\improper CentComm. hat"
+	name = "\improper Centcomm. hat"
```
Comment (BraveMole) [inline code/modules/clothing/head/misc.dm:4]:
```suggestion
	name = "\improper CentComm. hat"
```

File: html/changelogs/changelog20182.html [hunk 13]
```diff
@@ -563,7 +563,7 @@ <h3 class='author'>Surealistik updated:</h3>
 		</ul>
 		<h3 class='author'>Francinum updated:</h3>
 		<ul class='changes bgimages16'>
-			<li class='wip'>Brings CentComm slightly up to date, with a functional shuttle and a place for ERTs to return to.</li>
+			<li class='wip'>Brings Centcomm slightly up to date, with a functional shuttle and a place for ERTs to return to.</li>
```
Comment (BraveMole) [inline html/changelogs/changelog20182.html:566]:
```suggestion
			<li class='wip'>Brings CentComm slightly up to date, with a functional shuttle and a place for ERTs to return to.</li>
```

File: code/controllers/subsystem/vote.dm [hunk 14]
```diff
@@ -226,9 +226,9 @@ SUBSYSTEM_DEF(vote)
 					to_chat(usr, "<span class='warning'>The next ship map has already been selected.</span>")
 					return FALSE
 				var/list/maps = list()
+				if(!config.maplist)
+					return
 				for(var/map in config.maplist[SHIP_MAP])
-					if(!config.maplist)
-						return
```
Comment (BraveMole) [inline code/controllers/subsystem/vote.dm:231]: Ok so, i did a mistake in my other PR, it was on the wrong lines. 
This prevents runtime in integration test

**Review by TiviPlus — APPROVED** [review]
lgtm

**Review by spookydonut — APPROVED** [review]

**Pull request merged**